### DISCUSSION TITLE
check end processEntries with readdir

### DISCUSSION THREAD

**Discussion by stloc** [tauri-apps/tauri#5238] [Q&A]
Hello,

https://tauri.app/v1/api/js/fs/#readdir

how to execute code after recursive processEntries finished ?
log "an entry" displaying after log "end" 

```
const entries = await readDir('users', { dir: BaseDirectory.App, recursive: true });
processEntries(entries);
console.log(`end`);
// ?? check end processEntries with promise ?

function processEntries(entries) {
  for (const entry of entries) {
    console.log(`an entry`);
    if (entry.children) {
      processEntries(entry.children)
    }
  }
}
```

**Comment by FabianLars** [comment]
i think i don't understand your question tbh.

since `processEntries` is sync in your example there is no Promise involved (other than the one from `readDir`) and it will block on the `processEntries(entries);` line until it's done. Code after that will automatically wait until it's completed.

**Reply by stloc** [reply]
OK thanks. I just don't understand why log "an entry" displaying after log "end" in the browser, it must come from the browser behaviour
